File: Examples/ObjC/Sources/Analytics/AuthenticationManager.m
#import "AuthenticationManager.h"
#import "AppDelegate.h"

#pragma mark - User

@implementation User
@end

#pragma mark - AuthenticationManager

@implementation AuthenticationManager

- (void)userDidLogin:(User *)user {
  [AppDelegate.braze changeUser:user.identifier];
  BRZUser *brazeUser = AppDelegate.braze.user;
  [brazeUser setEmail:user.email];
  [brazeUser setDateOfBirth:user.birthday];
  [brazeUser setCustomAttributeWithKey:@"last_login_date"
                             dateValue:[NSDate date]];
}

@end

File: Examples/Sources/Analytics/AuthenticationManager.swift
import BrazeKit
import Foundation

class AuthenticationManager {

  struct User {
    let id: String
    let email: String
    let birthday: Date
  }

  func userDidLogin(_ user: User) {
    AppDelegate.braze?.changeUser(userId: user.id)
    let brazeUser = AppDelegate.braze?.user
    brazeUser?.set(email: user.email)
    brazeUser?.set(dateOfBirth: user.birthday)
    brazeUser?.setCustomAttribute(key: "last_login_date", value: Date())
  }

}
